File: Frontend/tempo-cheio-front/src/app/interfaces/TipoTarefa.ts
import { Tarefa } from './Tarefa';       
// import { TipoTarefa } from './TipoTarefa'; 


export interface TipoTarefa {
  id: number;
  nome: string;
  tarefa: Tarefa;       // Usa a interface Tarefa
  tipoTarefa: TipoTarefa; // Usa a interface TipoTarefa
}

File: Backend/tempo-cheio-back/src/main/java/com/tempo_cheio_back/model/TipoTarefa.java
package com.tempo_cheio_back.model;

import jakarta.annotation.Nonnull;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class TipoTarefa {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;

    @Nonnull
    private String nome;


}